apps/extension/src/core/managers/transaction.ts

```diff
@@ -41,28 +47,48 @@ class TransactionManager extends BaseManager<Transaction> {
   init<TInput extends Input>(
     input: TInput,
     origin: OriginData,
-    options: CompletionOptions<ModelID | string, TInput>
+    options: CompletionOptions<ModelID | string, TInput> | MediaOptions<ModelID> | ThreeDOptions<ModelID | string>
   ): Transaction {
     this._validateInput(input)
+  
+    // Extracting parameters common to all options
     const {
-      temperature,
-      maxTokens,
-      stopSequences,
       model,
-      numOutputs = 1
+      numOutputs = 1,
     } = options
+
+    let temperature: number | undefined
+    let maxTokens: number | undefined
+    let stopSequences: string[] | undefined
+    let numInferenceSteps: number | undefined
+
+    if ('temperature' in options || 'maxTokens' in options || 'stopSequences' in options) {
```

Comment:
nit: I think making some type helper methods would clean this up a bit:

"if isTextCompletionOptions(options) ...

function isTextCompletionOptions(opts): opts is CompletionOptions {
  ...(the code you have here)...
}"

Reply:
done!